wallet-new/src/Cardano/Wallet/WalletLayer/Types.hs

```diff
@@ -111,6 +119,103 @@ instance Buildable CreateAddressError where
     build (CreateAddressTimeLimitReached timeLimit) =
         bprint ("CreateAddressTimeLimitReached " % build) timeLimit
 
+------------------------------------------------------------
+-- Errors when dealing with Accounts
+------------------------------------------------------------
+
+data CreateAccountError =
+      CreateAccountError Kernel.CreateAccountError
+    | CreateAccountWalletIdDecodingFailed Text
+    -- ^ Decoding the parent's 'WalletId' from a raw 'Text' failed.
+    | CreateAccountTimeLimitReached TimeExecutionLimit
+    | CreateAccountFirstAddressGenerationFailed Kernel.CreateAddressError
+    -- ^ When trying to create the first 'Address' to go in tandem with this
+    -- 'Account', the generation failed.
+    deriving Eq
+
+-- | Unsound show instance needed for the 'Exception' instance.
+instance Show CreateAccountError where
+    show = formatToString build
+
+instance Exception CreateAccountError
```

Comment:
Rather than derive exception, does it make sense to have our own `data SomeWalletException = forall e. Exception e => SomeWalletException e` to define an exception hierarchy?

Reply:
Quite possibly -- however, as we have quite a legacy in this area (see our V1's `WalletError` type, which encapsulates old data layer errors) I have decided to not open the Pandora's box.. just yet 😉